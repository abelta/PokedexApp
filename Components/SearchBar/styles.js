import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  searchBar: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  back: {
    paddingLeft: 4,
    paddingRight: 4,
  },
  input: {
    flex: 1,
    color: 'grey',
    fontSize: 18,
    paddingLeft: 4,
    paddingRight: 4,
  },
  clear: {
    paddingLeft: 4,
    paddingRight: 4,
  },
});

export default styles;
