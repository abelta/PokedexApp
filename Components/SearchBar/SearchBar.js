import React from 'react';
import { Text, TextInput, View } from 'react-native';
import { Link } from 'react-router-native';
import styles from './styles';

const SearchBar = ({ style }) => (
  <View style={[styles.searchBar, style]}>
    <Link to="/"><Text>Home</Text></Link>
    <TextInput style={styles.input} />
  </View>
);

SearchBar.propTypes = {
  style: View.propTypes.style,
};

SearchBar.defaultProps = {
  style: {},
};

export default SearchBar;
