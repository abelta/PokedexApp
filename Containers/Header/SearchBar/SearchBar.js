import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TextInput } from 'react-native';
import { withRouter } from 'react-router-native';
import { SlideView, TouchableButton, TouchableLink } from '../../../Components';
import { Back as BackIcon, Clear as ClearIcon, Go as GoIcon } from '../../../Components/Icons';
import styles from './styles';

class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = { value: '' };
    this.onChangeText = this.onChangeText.bind(this);
    this.onClear = this.onClear.bind(this);
    this.onSubmitEditing = this.onSubmitEditing.bind(this);
  }

  onChangeText(value) {
    this.setState({ value: value.toLowerCase() });
  }

  onClear() {
    const value = '';
    this.setState({ value });
  }

  onSubmitEditing() {
    const { history } = this.props;
    const { value } = this.state;
    history.push(`/pokemon/${value}`);
  }

  render() {
    const { value } = this.state;
    const disabledButtons = value.length === 0;
    return (
      <SlideView style={styles.searchBar} toLeft >
        <TouchableLink style={styles.back} to="/" >
          <BackIcon />
        </TouchableLink>
        <TextInput
          autoCorrect={false}
          autoFocus
          multiline={false}
          onChangeText={this.onChangeText}
          onSubmitEditing={this.onSubmitEditing}
          placeholder="Find PokÃ©mon by name"
          placeholderTextColor="lightgrey"
          returnKeyType="search"
          selectTextOnFocus
          style={styles.input}
          underlineColorAndroid="transparent"
          value={value}
        />
        <TouchableLink
          disabled={disabledButtons}
          style={styles.go}
          to={`/pokemon/${value}`}
        >
          {!disabledButtons && <GoIcon />}
        </TouchableLink>
        <TouchableButton
          disabled={disabledButtons}
          onPress={this.onClear}
          style={styles.clear}
        >
          {!disabledButtons && <ClearIcon />}
        </TouchableButton>
      </SlideView>
    );
  }
}

SearchBar.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  history: PropTypes.object.isRequired,
};

export default withRouter(SearchBar);
