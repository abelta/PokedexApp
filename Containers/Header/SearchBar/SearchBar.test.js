import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-native';
import SearchBar from '.';

jest.mock('../../../Components/SlideView');

describe('SearchBar', () => {
  const tree = renderer.create(
    <MemoryRouter initialEntries={['/search']}>
      <SearchBar />
    </MemoryRouter>,
  ).toJSON();

  it('renders properly', () => {
    expect(tree).toMatchSnapshot();
  });
});
