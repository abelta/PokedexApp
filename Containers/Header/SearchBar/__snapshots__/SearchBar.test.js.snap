// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders properly 1`] = `null`;
